### ISSUE TITLE
No docstring popup in codecompletion is provided when immediatly calling a returned nested fuction

### ISSUE THREAD

**Issue by DeltaRazero** [microsoft/pylance-release#1990]
## Environment data

-   VSCode Version: 1.61.0-insider
-   Commit: c9e8266ee24f34107ff52d836a4e8f533afb93c5
-   Language Server version: v2021.10.2
-   OS and version: Windows 7 (Windows_NT x64 6.1.7601)
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.6.7

## Expected behaviour

When defining a nested function that will be returned to the user, a docstring popup is provided in the code editor/completion when directly calling the returned function after calling top level function.

## Actual behaviour

When defining a nested function that will be returned to the user, NO docstring popup is provided in the code editor/completion when directly calling the returned function after calling top level function.

(See code snippet example underneath)
![actual behaviour](https://user-images.githubusercontent.com/26445643/139120172-631e808f-bfc7-4030-bc09-3eb28fa1306a.png)

But if you do it like this it works:
![image](https://user-images.githubusercontent.com/26445643/139120725-7db6ddec-c6dd-4877-afdc-cca8cf1c1aa1.png)
![image](https://user-images.githubusercontent.com/26445643/139120780-03997522-022c-44e2-9be2-30d795c81031.png)

## Code Snippet / Additional information

```python
def foo(template_var: int):
    """Some function description.

    Parameters
    ----------
    template_var : int
        Some template variable.
    """

    def bar(input_var: int) -> int:
        """Some templated function description.

        Parameters
        ----------
        input_var : int
            Some input variable.

        Returns
        -------
        int
            Output value.
        """
        return template_var + input_var

    return bar


# Yields no docstring popup
foo(5)(5)

# But if you store the result into a var...
a = foo(5)
# it does show up
a(5)
```

**Comment by DeltaRazero** [comment]
Any update on this issue yet?

**Comment by bschnurr** [comment]
This issue has been fixed in version 2022.7.31, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2022731-13-july-2022)
